src/main/java/io/odpf/kafkaclient/PropertiesNotSetException.java

```diff
@@ -1,4 +1,21 @@
 package io.odpf.kafkaclient;
 
-public class PropertiesNotSetException extends RuntimeException{
+import java.util.logging.ConsoleHandler;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+
+public class PropertiesNotSetException extends RuntimeException {
+
+    @Override
+    public void printStackTrace() {
+        Logger log = Logger.getLogger("client.logger");
+        log.setLevel(Level.ALL);
+        ConsoleHandler handler = new ConsoleHandler();
+
+        handler.setLevel(Level.ALL);
+        handler.setFormatter(new SimpleFormatter());
+        log.addHandler(handler);
+        log.info("Properties of Client Factory not set");
```

Comment:
why info ? The log level should be error. Read more on logger.
Also why this much repetitive code. 
Also, why are you overriding stack trace printing?

Reply:
Make custom Exceptions minimal. Have a look at one of the Exceptions we use in prod. https://github.com/odpf/firehose/blob/f62725965f06569817f68cc2a669036fbc6f26fb/src/main/java/io/odpf/firehose/exception/DeserializerException.java#L6

Reply:
Deleted the override and implemented constructor for exception classes.